### ISSUE TITLE
Auto loaded extensions

### ISSUE THREAD

**Issue by sandman7920** [sqlitebrowser/sqlitebrowser#2467]
Details for the issue
--------------------

Preferences dialog doesn't save the extensions list correctly.

https://github.com/sqlitebrowser/sqlitebrowser/blob/master/src/PreferencesDialog.cpp#L270
```cpp
ui->listExtensions->findItems(QString("*"), Qt::MatchWrap | Qt::MatchWildcard) // returns empty list
```
Maybe this is a Qt 5.15 bug

My workaround
```cpp
    for(int i = 0, count = ui->listExtensions->count(); i < count; ++i)
        extList.append(ui->listExtensions->item(i)->text());
```

Useful extra information
-------------------------
![image](https://user-images.githubusercontent.com/3589417/97962421-d7b63880-1dbd-11eb-9924-b250eefe0d61.png)

#### What operating system are you using?

- [ ] Windows: ( _version:_ ___ )
- [x] Linux: (Arch Linux rolling)
- [ ] macOS: ( _version:_ ___ )
- [ ] Other: ___

#### What is your DB4S version?

- [x] 3.12.0
- [ ] 3.11.x
- [ ] Other: ___

**Comment by justinclift** [comment]
Oh, that sounds like a bug.  Thanks for pointing it out @sandman7920. :smile:

**Comment by sandman7920** [comment]
I have made a test with a new empty project and ```ui->listExtensions->findItems(QString("*"), Qt::MatchWrap | Qt::MatchWildcard)``` returns full StringList

Very strange

**Comment by sandman7920** [comment]
Just for test I compiled with Qt 5.12.9 and 5.15.1 on Windows.
5.12.9 -> OK
5.15.1 -> Failed (empty list)

**Comment by justinclift** [comment]
Awesome!  That's really useful info.  Thanks heaps @sandman7920. :grin:

**Comment by justinclift** [comment]
@MKleusberg We might have to use Qt 5.15.1 for our macOS builds, in order to have the binaries "Notarised" by Apple.

Would you be ok to look at the concept patch above?

**Comment by sandman7920** [comment]
Squashing this bug is a must.
Currently, all official Linux packages on rolling update distros (at least arch based) are affected.

**Comment by justinclift** [comment]
Ahhh, that makes sense.  Thanks for the info @sandman7920. :smile:

**Comment by MKleusberg** [comment]
Thanks, @sandman7920! I have adapted your "workaround" (which in my opinion is the correct way to do this) and pushed it to the master branch. Can we do anything else to assist fixing this in the Arch based distros?

**Comment by justinclift** [comment]
@MKleusberg Would this apply ok to the `v.3.12.x` branch?

Asking because we're using Qt 5.15.1 for our macOS build (needed for Apple's "Notarising"), so this bug is likely happening there too.  And we need to create new v3.12.x builds anyway, to include the missing `fileio` extension.

**Comment by MKleusberg** [comment]
@justinclift Sure. I've just cherry-picked this commit.

**Comment by justinclift** [comment]
Excellent, thanks heaps @MKleusberg. :smile:

**Comment by sandman7920** [comment]
@MKleusberg
>Can we do anything else to assist fixing this in the Arch based distros?

After a new release is published 3.12.2 everything should be fine.
